content/blog/2020-04-dplyr-1-0-0-colwise/index.Rmarkdown

```diff
@@ -0,0 +1,173 @@
+
+starwars %>% 
+  group_by(homeworld) %>% 
+  filter(n() > 1) %>% 
+  summarise(across(is.numeric, mean, na.rm = TRUE), n = n())
+```
+## Other cool features
+
+You'll find a lot more about `across()` in [`vignette("colwise")`](https://dplyr.tidyverse.org/dev/articles/colwise.html). There are three cool features you might be particularly interested in:
+
+* You can use it with [multiple summary functions](https://dplyr.tidyverse.org/dev/articles/colwise.html#multiple-functions).
+
+* You can use it with [any dplyr verb](https://dplyr.tidyverse.org/dev/articles/colwise.html#other-verbs).
+
+* If needed, you can access the name of the column currently being processed
+  with [`cur_column()`](https://dplyr.tidyverse.org/dev/articles/colwise.html#current-column).
+
+## Why `across()`? 
+
+If you've tackled this problem with an older version of dplyr, you might've used one of the functions had an `_if`, `_at`, or `_all` suffix. These functions solved a pressing need and are used by many people, but are now superseded. This means that they'll stay around, but will only recieve critical bug fixes. 
+
+Why did we decide to move away from these functions in favour of `across()`?
+
+1.  `across()` makes it possible to compute useful summaries that were 
+    previously impossible. For example, it's now trivial to summarise
+    numeric vectors with one function, factors with another, and still 
+    compute the number of rows in each group:
+
+    ```{r, eval = FALSE}
+    df %>%
+      group_by(g1, g2) %>% 
+      summarise(
+        across(is.numeric, mean), 
+        across(is.factor, nlevels),
+        n = n(), 
+      )
+    ```
+
+2.  `across()` reduces the number of functions that dplyr needs to provide. 
+    This makes dplyr easier for you to use (because there are fewer functions 
+    to remember) and easier for us to develop (since we only need to implement 
+    one function for each new verb, not four).
+
+3.  `across()` unifies `_if` and `_at` semantics, allowing combinations that 
+    used to be impossible. For example, you can now transform all numeric 
+    columns whose name begins with "x": `across(is.numeric & starts_with("x"))`.
+
+4.  `across()` doesn't need `vars()`. The `_at()` functions are the only place 
+    in dplyr where you have to use `vars()`, which makes them unusual, 
+    and hence harder to learn and remember.
+
+Why did it take it long to discover `across()`? Surprisingly, the key idea that makes `across()` works came out of our low-level work on the [vctrs](http://vctrs.r-lib.org/) package, where we learnt that you can have a column of a data frame that is itself a data frame. It's a bummer that we had a few false starts before we discovered `across()`, but even with hindsight, I don't see how we could've skipped the intermediate steps.
+
+## How do you convert existing code?
+
+Fortunately, if you have code that uses the `_if`, `_at`, or `_all()` functions, it's generally straightforward to update:
```

Comment:
```suggestion
If you have code that uses the `_if`, `_at`, or `_all()` functions, it's generally straightforward to update:
```

Reply:
To me, "Fortunately" makes it sound more like updating might be strictly necessary (which it isn't, given what we mean by superseded). I think it's good to be extra careful here, since there's still a tenuous understanding of the term out there.

Reply:
Oh, I see you have extra reassurance later, but I still think we can axe fortunately.

Reply:
Yeah, I'm struggled with this — I want to imply that it's easy, but you don't have to do it.  Maybe: If you want to update your existing code ...